kingpin/actors/rightscale/base.py

```diff
@@ -213,6 +213,38 @@ def _delete_resource_tags(self, resource, tags):
         yield self._client.delete_resource_tags(resource, tags)
 
     @gen.coroutine
+    def _ensure_tags(self, res, tags):
+        """Ensures that a set of tags are applied to a RightScale resource.
+
+        Args:
+            res: The resource object itself
+            tags: A list of tags to apply
+        """
+        if isinstance(tags, basestring):
+            tags = [tags]
+
+        if not res.href:
+            # Must be a mocked-out MCI, meaning its brand new, meaning it has
+            # no tags. Definitely set them.
+            yield self._add_resource_tags(resource=res, tags=tags)
+            self.changed = True
```

Comment:
Don't know what this is, but probably should be in inside of `_add_resource_tags`